### ISSUE TITLE
Unclear error message when setting allowed-connection=specified and not providing any peer-key

### ISSUE THREAD

**Issue by matthewdarwin** [AntelopeIO/spring#155]
Unclear error message when setting `allowed-connection=specified` and not providing any peer-key

```
appbase: exception thrown during plugin "" initialization.
Incorrect plugin configuration
```

**Comment by spoonincode** [comment]
What version are you seeing that on? I tried spring 1.0.0 beta1 and leap 5.0.2, both provided a helpful error message.

```
nodeos --allowed-connection=specified
...
warn  2024-05-17T22:07:00.239 nodeos    net_plugin.cpp:4406           plugin_initialize    ] 3110006 plugin_config_exception: Incorrect plugin configuration
At least one peer-key must accompany 'allowed-connection=specified'
    {}
    nodeos  net_plugin.cpp:4370 plugin_initialize

appbase: exception thrown during plugin "eosio::net_plugin" initialization.
Incorrect plugin configuration
```
but I do see something like looks like #138 on spring beta, since I end up with,
```
warn  2024-05-17T22:07:00.244 nodeos    fork_database.cpp:722         close                ] fork_database is in a bad state when closing; not writing out '/home/spoon/.local/share/eosio/nodeos/data/blocks/reversible/fork_db.dat', legacy_valid=false, savanna_valid=false
```

**Comment by heifner** [comment]
```
warn  2024-05-17T22:07:00.244 nodeos    fork_database.cpp:722         close                ] fork_database is in a bad state when closing; not writing out '/home/spoon/.local/share/eosio/nodeos/data/blocks/reversible/fork_db.dat', legacy_valid=false, savanna_valid=false
```
We should fix that warning, there is nothing wrong there.

**Comment by heifner** [comment]
Resolved by #164
